server/tests/test_pipeline.py

```diff
@@ -17,6 +22,30 @@ class Case(BaseModel):
     expected_result: list
 
 
+def jinja_renderer(query: dict | list[dict] | tuple | str, parameters: dict):
+    if isinstance(query, dict):
+        return {key: jinja_renderer(value, parameters) for key, value in deepcopy(query).items()}
+    elif isinstance(query, list):
+        rendered_query = [jinja_renderer(elt, parameters) for elt in deepcopy(query)]
+        return rendered_query
+    elif isinstance(query, tuple):
+        return tuple(jinja_renderer(value, parameters) for value in deepcopy(query))
+    elif isinstance(query, str):
+        # Replace param templating with jinja templating:
+        query = re.sub(RE_PARAM, r"{{ \g<1> }}", query)
+
+        # Add quotes to string parameters to keep type if not complex
+        clean_p = deepcopy(parameters)
+        if is_jinja_alone(query):
+            env = NativeEnvironment()
+        else:
+            env = Environment()
+
+        return env.from_string(query).render(clean_p)
+    else:
+        return query
```

Comment:
```suggestion
def jinja_renderer(query: Any, parameters: dict):
    return NativeEnvironment().from_string(json.dumps(query)).render(parameters)
```

Reply:
This doesn't work with datetime types variables !
please did you tested it ?
+ I had to add a DatetimeEncoder to have this implementation :
 ```python
class DateTimeEncoder(json.JSONEncoder):
    def default(self, obj):
        if isinstance(obj, datetime):
            return obj.isoformat()
        return super().default(obj)


def jinja_renderer(query: Any, parameters: dict) -> Any:
    query_rendered = (
        NativeEnvironment().from_string(json.dumps(query, cls=DateTimeEncoder)).render(parameters)
    )
    if isinstance(query_rendered, str):
        query_rendered = json.loads(query_rendered)
    return query_rendered
```
even with this change, the variable parameters are changed from integers to strings (like 32 -> '32' when rendered), do you have any tips please ?

Reply:
Hmmm, my bad, I have overestimated `NativeEnvironment's` capabilities, it seems to do some weird guessing in some situations. Let's stick with the old form then. I've simplified it because we don't need to support different templating types: https://github.com/ToucanToco/weaverbird/pull/1763/commits/fe20067318f5c377b890bd0efe3006780769e586

Reply:
👍🏽

Reply:
Thanks for the update, if the whole PR is okay for you we can then merge this and make some bumps, WDYT ?

Reply:
woops yes sorry forgot the approval